File: storydoc-backend/src/main/java/io/storydoc/server/ui/app/screendesign/SDContainerDTO.java
package io.storydoc.server.ui.app.screendesign;

import io.storydoc.server.ui.domain.screendesign.SDComponentId;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class SDContainerDTO {
    private String label;
    private SDComponentId id;
    private List<SDComponentTypeSelectionDTO> children;
}

File: storydoc-frontend/src/app/api/models/sd-container-dto.ts
/* tslint:disable */
/* eslint-disable */
import { SdComponentId } from './sd-component-id';
import { SdComponentTypeSelectionDto } from './sd-component-type-selection-dto';
export interface SdContainerDto {
  children?: Array<SdComponentTypeSelectionDto>;
  id?: SdComponentId;
  label?: string;
}
